### ISSUE TITLE
Arango Search wrong equals behaviour on arrays?

### ISSUE THREAD

**Issue by aveiros** [arangodb/arangodb#8294]
## My Environment

* __ArangoDB Version__:        3.4.4-PREVIEW
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:    ArangoDB Starter in Docker
* __Infrastructure__:               own
* __Operating System__:        MacOS 10.13.4
* __Total RAM in your machine__:        32Gb
* __Disks in use__:        SSD
* __Used Package__:              Docker - official Docker library

In Arango Search, I wonder if array comparison on strings is correct.
I have a entities collection with the following document:
```
{
  "type": "address",
  "id": "address1",
  "labels": [ "A", "B" ]
}
```

I have a entities_view as follow
```
{
  "writebufferSizeMax": 33554432,
  "consolidationPolicy": {
    "type": "bytes_accum",
    "threshold": 0.10000000149011612
  },
  "globallyUniqueId": "hC77195588A3E/350",
  "id": "350",
  "writebufferActive": 0,
  "consolidationIntervalMsec": 60000,
  "cleanupIntervalStep": 10,
  "links": {
    "entities": {
      "analyzers": [
        "identity"
      ],
      "fields": {},
      "includeAllFields": true,
      "storeValues": "id",
      "trackListPositions": false
    }
  },
  "type": "arangosearch",
  "writebufferIdle": 64
}
```

Running the following query:

```
FOR d in entities_view SEARCH d.labels == 'A' && d.labels == 'B' RETURN d
```

Expected Result: Address 1 Document
Actual Result: Empty

Workarounds provided by the Arango Team
Workaround 1 

```
LET v1 = V8("A")
LET v2 = V8("B")
FOR d in entities_view SEARCH d.labels == v1 && d.labels == v2 RETURN d
```

Workaround 2

```
LET v1 = NOOPT("A")
LET v2 = NOOPT("B")
FOR d in entities_view SEARCH d.labels == v1 && d.labels == v2 RETURN d
```

**Comment by maxkernbach** [comment]
Hi @aveiros,

version 3.4.5, which includes a fix for this issue, has just been released.
Could you please give it a try?

The latest release is available in the [download section](https://www.arangodb.com/download-major/).

**Comment by aveiros** [comment]
@maxkernbach can confirm this is fixed on 3.4.5.

Thank you.
